docs/tdag.md

```diff
@@ -0,0 +1,105 @@
+# TaintDAG file format
+
+The Taint Directed Acyclic Graph (TaintDAG, TDAG) file format is tailored to facilitate fast recording of taint operations.
+
+It is a binary file format based on sparse files. It consists of a header and a number of subsections. The subsections stores information about:
+* taint sources, filename and offset information
+* taint output log, tainted values written to an output file
+* taint graph, the graph of how taint values are unioned from source taint and other unions.
+
+Because of the sparse layout it is very well suited for mmap'ing directly into the instrumented process address space.
+
+## Taint Sources, Unions and Ranges
+Whenever data is read from an input file, the data entering the program is labeled as source taint. Information about which file and at what offset is kept. This is the only way taint can originate in a program.
+
+Whenever an operation is performed where one or more operands are involved a union or range value is created. If the taint labels are adjacent (number wise), e.g. two consecutive source taint bytes, a range is created. If the labels are not adjacent a union of the two labels is created. Unions and ranges occupy the same amount of storage. The main difference is that a range can be extended to become a larger range.
```

Comment:
Probably be better if we specify that the operations we have in mind are operations of the instrumented program and the operands need to be tainted for a union or range to be created. Is that right?

Reply:
Hmm. Maybe this whole paragraph needs insight into how we do taint analysis. So the previous comment might not be relevant.